### PR TITLE
Upgrade NestJS one major up (9), but not the latest

### PR DESCRIPTION
**Pull request by paazmaya**
... and fix few issues that are found on the way.

### CODE REVIEW

**Comment by paazmaya** [comment]
Took https://github.com/darraghoriordan/eslint-plugin-nestjs-typed in to use, as it might help in catching issues with the major version upgrades in NestJS.

**Comment by paazmaya** [comment]
@pashidlos running `npm start` fails due to a `isEnum` check. Would you have better understanding about where it is coming from and how to fix it?

File: .eslintrc.js [hunk 1]
```diff
@@ -4,10 +4,11 @@ module.exports = {
     project: 'tsconfig.json',
     sourceType: 'module',
   },
-  plugins: ['@typescript-eslint/eslint-plugin'],
+  plugins: ['@typescript-eslint/eslint-plugin', '@darraghor/nestjs-typed'],
```
Comment (pashidlos) [inline .eslintrc.js:7]: move this part out in favor of https://github.com/Visual-Regression-Tracker/backend/pull/233/files

Comment (paazmaya) [reply .eslintrc.js:7]: I will rebase on top of master once there is updates to master...

File: .eslintrc.js [hunk 2]
```diff
@@ -20,5 +20,6 @@ module.exports = {
     '@typescript-eslint/interface-name-prefix': 'off',
     '@typescript-eslint/explicit-function-return-type': 'off',
     '@typescript-eslint/no-explicit-any': 'off',
+    '@typescript-eslint/no-unused-vars': 'off', // False positives from decorators
```
Comment (pashidlos) [inline .eslintrc.js:23]: would it be more safe to add ignore specifically to such places instead of switching off the check?

**Review by pashidlos — APPROVED** [review]
LGTM

File: .eslintrc.js (hunk 2 above)
Comment (paazmaya) [reply .eslintrc.js]: Looks like this was no longer needed

**Pull request merged**